From cf822e1dd2e5d159d9d29be6cae7b6027b13809e Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Sat, 13 Aug 2022 21:43:54 +0900
Subject: [PATCH 2/4] comp.c: add imported for maybe_garbage_collect

---
 src/comp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/comp.c b/src/comp.c
index 346771c91..5ac87c627 100644
--- a/src/comp.c
+++ b/src/comp.c
@@ -671,6 +671,7 @@ #define NUM_CAST_TYPES 15
     helper_unwind_protect,
     specbind,
     /*garbage_collect, *//*maybe_gc,*/
+    maybe_garbage_collect,
     maybe_quit };
 
 
@@ -2767,7 +2768,7 @@ #define ADD_IMPORTED(f_name, ret_type, nargs, args)			       \
   args[0] = args[1] = comp.lisp_obj_type;
   ADD_IMPORTED (specbind, comp.void_type, 2, args);
 
-  ADD_IMPORTED (maybe_gc, comp.void_type, 0, NULL);
+  ADD_IMPORTED (maybe_garbage_collect, comp.void_type, 0, NULL);
 
   ADD_IMPORTED (maybe_quit, comp.void_type, 0, NULL);
 
-- 
2.37.0.windows.1

