From 128afb9b6cd6fa2dfd59949eeaead1eb23a6cef0 Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Tue, 1 Apr 2025 10:41:46 -0700
Subject: [PATCH 10/11] dump_map_file_w32: fix bad merge endif

---
 src/pdumper.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/pdumper.c b/src/pdumper.c
index 9e46f32bb..0c0b308c0 100644
--- a/src/pdumper.c
+++ b/src/pdumper.c
@@ -4885,8 +4885,9 @@ dump_anonymous_release (void *addr, size_t size)
   emacs_abort ();
 #endif
 }
+#endif /* not HAVE_MPS */
 
-#elif VM_SUPPORTED == VM_MS_WINDOWS && !defined HAVE_MPS
+#if VM_SUPPORTED == VM_MS_WINDOWS && !defined HAVE_MPS
 static void *
 dump_map_file_w32 (void *base, int fd, off_t offset, size_t size,
 		   enum dump_memory_protection protection)
-- 
2.46.0.windows.1

