From 118e5414edef16f701ab29fc54149b92eaf9ad02 Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Thu, 14 Sep 2023 23:19:25 +0900
Subject: [PATCH 01/24] .el.elc: ignore byte compile errors


diff --git a/Makefile.in b/Makefile.in
index e150dd16d7d..8a285417097 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -757,7 +757,7 @@ install-arch-indep:
 	[ -z "${GZIP_PROG}" ] || { \
 	  echo "Compressing *.el etc. ..." && \
 	  cd "$(DESTDIR)${lispdir}" && \
-	  for f in `find . -name "*.elc" -print | sed 's/.elc$$/.el/'`; do \
+	  for f in `find . -name "*.elc" ! -name "*loaddefs.elc" -print | sed 's/.elc$$/.el/'`; do \
 	    ${GZIP_PROG} -9n "$$f"; \
 	  done; \
 	  ${GZIP_PROG} -9n "../etc/publicsuffix.txt"; \
diff --git a/lisp/Makefile.in b/lisp/Makefile.in
index 1142147fa8a..1045f5569b1 100644
--- a/lisp/Makefile.in
+++ b/lisp/Makefile.in
@@ -323,17 +323,17 @@ .SUFFIXES:
 # which uses it to avoid displaying certain messages which might be
 # irritating/misleading during a bootstrap.
 .el.elc:
-	$(AM_V_ELC)$(emacs) $(BYTE_COMPILE_FLAGS) \
+	-$(AM_V_ELC)$(emacs) $(BYTE_COMPILE_FLAGS) \
 	-l comp -f batch-byte-compile $<
 	TZ=UTC0 touch -t 197001010000 $@
 else
 .el.elc:
-	$(AM_V_ELC)$(emacs) $(BYTE_COMPILE_FLAGS) \
+	-$(AM_V_ELC)$(emacs) $(BYTE_COMPILE_FLAGS) \
 	-l comp -f batch-byte+native-compile $<
 endif
 else
 .el.elc:
-	$(AM_V_ELC)$(emacs) $(BYTE_COMPILE_FLAGS) -f batch-byte-compile $<
+	-$(AM_V_ELC)$(emacs) $(BYTE_COMPILE_FLAGS) -f batch-byte-compile $<
 endif
 
 .PHONY: compile-first compile-main compile compile-always
-- 
2.54.0.windows.1

