From ba673f2ea59040a5b788375ba5c72fb12f2384e8 Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Wed, 30 Apr 2025 22:27:07 -0700
Subject: [PATCH 08/24] makefile: dont build leim


diff --git a/lisp/Makefile.in b/lisp/Makefile.in
index 1045f5569b1..91255073f86 100644
--- a/lisp/Makefile.in
+++ b/lisp/Makefile.in
@@ -132,12 +132,12 @@ SUBDIRS_SUBDIRS =
 
 # cus-load, finder-inf and autoloads are not explicitly requested by
 # anything, so we add them here to make sure they get built.
-all: compile-main $(lisp)/cus-load.el $(lisp)/finder-inf.el generate-ja-dic \
+all: compile-main $(lisp)/cus-load.el $(lisp)/finder-inf.el \
 	org-manuals autoloads
 
 PHONY_EXTRAS =
 .PHONY: all custom-deps finder-data autoloads update-subdirs $(PHONY_EXTRAS) \
-	generate-ja-dic org-manuals
+	org-manuals
 
 # custom-deps and finder-data both used to scan _all_ the *.el files.
 # This could lead to problems in parallel builds if automatically
@@ -406,7 +406,7 @@ .PHONY:
 
 ## make -C ../admin/unidata all should be here, but that would race
 ## with ../src.  See comments above for loaddefs.
-gen-lisp: leim semantic
+gen-lisp: semantic
 
 # (re)compile titdic-cnv before recursing into `leim` since its used to
 # generate some of the Quail source files from tables.
-- 
2.54.0.windows.1

