From 14fca665d28bd92af0ccd58e4d11c917256f4f38 Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Fri, 23 Aug 2024 04:51:47 -0700
Subject: [PATCH 11/24] Stop using legacy -ffat-lto-objects option in LTO
 builds


diff --git a/configure.ac b/configure.ac
index 09bd782240f..db5622aea7f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1995,13 +1995,6 @@ AC_DEFUN
 	 # command, so plugin name is appended to ARFLAGS.
 	 ARFLAGS="cru --plugin $GOLD_PLUGIN"
 	 RANLIB="$RANLIB --plugin $GOLD_PLUGIN"
-      else
-        dnl The following is needed for GCC 4.9.0.  The GCC 4.9.0 release notes
-        dnl suggest that instead of -ffat-lto-objects we should use gcc-ar and
-        dnl gcc-ranlib in place of ar and ranlib, but gcc-ar makes /usr/bin/ar
-        dnl dump core on Fedora 20, so play it safe for now.
-        gl_COMPILER_OPTION_IF([-ffat-lto-objects],
-          [CFLAGS="$CFLAGS -ffat-lto-objects"])
       fi
    fi
 fi)
-- 
2.54.0.windows.1

