diff --git a/configure.ac b/configure.ac
index b3276753cdb57005cc0a56f149a4ac89c434634f..a1b483213d34992060ace119368f63362dcb0953 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1562,7 +1562,7 @@ AC_DEFUN
   ;;
 
   # MinGW64
-  x86_64-*-* )
+  x86_64-*-* | aarch64-*-* )
     case "${canonical}" in
       *-mingw* )
 		opsys=mingw32
@@ -2286,6 +2286,7 @@ AC_DEFUN
 if test "$opsys" = "mingw32"; then
   case "$canonical" in
     x86_64-*-mingw*) C_SWITCH_SYSTEM="-mtune=generic" ;;
+    aarch64-*-mingw*) C_SWITCH_SYSTEM="-mtune=cortex-a53" ;;
     *) C_SWITCH_SYSTEM="-mtune=pentium4" ;;
   esac
 fi
@@ -3083,6 +3084,7 @@ AC_DEFUN
   EMACSRES="emacs.res"
   case "$canonical" in
     x86_64-*-*) EMACS_MANIFEST="emacs-x64.manifest" ;;
+    aarch64-*-*) EMACS_MANIFEST="emacs-ARM64.manifest" ;;
     *) EMACS_MANIFEST="emacs-x86.manifest" ;;
   esac
   dnl Construct something of the form "24,4,0,0" with 4 components.
@@ -7564,7 +7566,7 @@ AC_DEFUN
    ## If the values of -image-base are modified, the corresponding
    ## values of DEFAULT_IMAGE_BASE in w32fns.c should be kept in sync.
    case "$canonical" in
-     x86_64-*-*) LD_SWITCH_SYSTEM_TEMACS="-Wl,-stack,0x00800000 -Wl,-heap,0x00100000 -Wl,-image-base,0x400000000 -Wl,-entry,__start -Wl,-Map,./temacs.map" ;;
+     x86_64-*-* | aarch64-*-*) LD_SWITCH_SYSTEM_TEMACS="-Wl,-stack,0x00800000 -Wl,-heap,0x00100000 -Wl,-image-base,0x400000000 -Wl,-entry,__start -Wl,-Map,./temacs.map" ;;
      *) LD_SWITCH_SYSTEM_TEMACS="-Wl,-stack,0x00800000 -Wl,-heap,0x00100000 -Wl,-image-base,0x01000000 -Wl,-entry,__start -Wl,-Map,./temacs.map" ;;
    esac
    ;;
