From 5181592d2b678b235046ad3a769d6a43263b901b Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Fri, 19 Jun 2026 18:00:59 -0700
Subject: [PATCH 13/24] setup_w32_kbhook: ignore it


diff --git a/src/w32console.c b/src/w32console.c
index 30a1f65f43f..9aaa011d9a4 100644
--- a/src/w32console.c
+++ b/src/w32console.c
@@ -1134,7 +1134,7 @@ initialize_w32_display (struct terminal *term, int *width, int *height)
   EnumThreadWindows (GetCurrentThreadId (), find_ime_window, (LPARAM) &hwnd);
 
   /* Set up the keyboard hook.  */
-  setup_w32_kbdhook (hwnd);
+  /* setup_w32_kbdhook (hwnd); */
 
   /* Set current_tty to the tty of this terminal */
   current_tty = term->display_info.tty;
diff --git a/src/w32fns.c b/src/w32fns.c
index 3e95a48ac80..62e02916d42 100644
--- a/src/w32fns.c
+++ b/src/w32fns.c
@@ -5723,7 +5723,7 @@ #define WM_TOUCH 576
 
 #ifdef WINDOWSNT
     case WM_CREATE:
-      setup_w32_kbdhook (hwnd);
+      /* setup_w32_kbdhook (hwnd); */
       goto dflt;
 #endif
 
-- 
2.54.0.windows.1

